### PR TITLE
feat: STAKE-802 integrate claim sdk method

### PR DESCRIPTION
**Pull request by Matt561**
## **Description**
This PR adds native mobile support for claiming pool staked ETH.

## **Related issues**

Jira Ticket: [FE] [Unstake flow] Integrate unstake method from sdk ([link](https://consensyssoftware.atlassian.net/browse/STAKE-803))

## **Manual testing steps**

prerequisite: if you don't have any Ethereum unstaked to claim, you'll need to unstake some ETH first.

1. Add `export MM_POOLED_STAKING_UI_ENABLED=true` to `.js.env` file.
2. Click on Ethereum from the asset list page
3. Click on "Claim ETH" text button
4. Click "Confirm" on the transaction-controller confirmation screen.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/Stake/hooks/usePoolStakedClaim/index.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+  const txParams = generateClaimTxParams(
+    activeAccountAddress,
+    poolStakingContract.contract.address,
+    data,
+    chainId,
+    gasLimit.toString(),
+  );
+
+  return addTransaction(txParams, {
```
Comment (amitabh94) [inline app/components/UI/Stake/hooks/usePoolStakedClaim/index.ts:85]: Shouldn't this be awaited?

File: app/components/UI/Stake/hooks/usePoolStakedClaim/index.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
```
Comment (amitabh94) [inline app/components/UI/Stake/hooks/usePoolStakedClaim/index.ts:42]: Shouldn't this be awaited ?

**Review by amitabh94 — COMMENTED** [review]
Raised some questions but otherwise LGTM!

Also, it may be tackled in a separate PR or same but I noticed that when you try to edit the gas fees on the confirmation modal it gives out an error.

File: app/components/UI/Stake/hooks/usePoolStakedClaim/index.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+const attemptPoolStakedClaimTransaction =
+  (poolStakingContract: PooledStakingContract) =>
+  async (activeAccountAddress: string, pooledStakesData: PooledStake) => {
+    try {
+      if (pooledStakesData.exitRequests.length === 0) return;
+
+      const isMultiCallClaim = pooledStakesData.exitRequests.length > 1;
+
+      return isMultiCallClaim
+        ? attemptMultiCallClaimTransaction(
```
[the remark this answers is not in the document]
Comment (Matt561) [reply app/components/UI/Stake/hooks/usePoolStakedClaim/index.ts:100]: Yeah it makes sense here. For the `addTransaction` calls I was relying on the caller to handle the error originally but I can add `await` for those too.

File: app/components/UI/Stake/hooks/usePoolStakedClaim/index.ts (hunk 1 above)
Comment (Matt561) [reply app/components/UI/Stake/hooks/usePoolStakedClaim/index.ts:85]: I was originally expecting the caller to `.catch` but will add it here.

File: app/components/UI/Stake/hooks/usePoolStakedClaim/index.ts (hunk 2 above)
Comment (Matt561) [reply app/components/UI/Stake/hooks/usePoolStakedClaim/index.ts:42]: I was originally expecting the caller to `.catch` but will add it here.

**Comment by Matt561** [comment]
> Raised some questions but otherwise LGTM!
> 
> Also, it may be tackled in a separate PR or same but I noticed that when you try to edit the gas fees on the confirmation modal it gives out an error.

I noticed this too. I haven't looked into it yet though. I wasn't sure if it was something that needed fixing on the confirmation side.

File: app/components/UI/Stake/hooks/usePoolStakedClaim/index.ts [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+  const gasLimit = await poolStakingContract.estimateMulticallGas(
+    multiCallData,
+    activeAccountAddress,
+  );
+
+  const { data, chainId } =
+    await poolStakingContract.encodeMulticallTransactionData(
+      multiCallData,
+      activeAccountAddress,
+      { gasLimit, gasBufferPct: 30 }, // 30% buffer
```
Comment (amitabh94) [inline app/components/UI/Stake/hooks/usePoolStakedClaim/index.ts:31]: Should this gas buffer be 0.3 instead of 30 ?

Comment (Matt561) [reply app/components/UI/Stake/hooks/usePoolStakedClaim/index.ts]: You're correct. I've actually just removed this so we can rely on the default buffer used by the SDK for now. The default buffer is already 30%.

**Review by amitabh94 — APPROVED** [review]

**Pull request merged**